elements/lisk-p2p/src/p2p.ts

```diff
@@ -349,26 +333,16 @@ export class P2P extends EventEmitter {
 		};
 
 		this._handlePeerInfoUpdate = (peerInfo: P2PPeerInfo) => {
-			try {
+			const existingPeer = this._peerBook.getPeer(peerInfo);
```

Comment:
Why do we change this back? The idea is to not to check if the peer exists twice. `addPeer()` throws an error if the peer exists.

Reply:
The usage of `try catch` here is not correct. It is easier to understand by simply checking if peer exists in peerBook.

Reply:
Then you check twice...

Reply:
You only check once if it exists. The overhead of a single `if` check is worth it if you compare the ease of readability.

Reply:
Then we need to follow the same approach everywhere. There are other places where we proceed as I've suggested.

Reply:
I added `peerBook.hasPeer` for faster checking and applied this approach for the existing `try catch` related to `peerBook.addPeer`.